components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1161,6 +1165,31 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         }
     }
 
+    async getHeadlessLog(instanceId: string): Promise<HeadlessLogSources> {
+        const user = this.checkAndBlockUser();
+        const span = opentracing.globalTracer().startSpan("getHeadlessLog");
+        const context: LogContext = { userId: user.id, instanceId };
+        log.info(context, 'getHeadlessLog', { instanceId });
```

Comment:
isn't this stuffed into `checkAndBlockUser`? I'd prefer to keep the usage logging contained in there s.t. we have an easier time enabling/disabling it in the future.

Reply:
cargo cult, will remove